docs/connectors/matrix.md

```diff
@@ -6,25 +6,26 @@ Maintained by [@SolarDrew](https://github.com/SolarDrew).
 
 ## Requirements
 
-To use this connector you will need to have a Matrix account, and login using your Matrix username (mxid) and access token. You can find the access token in the **Advanced** section of the **Help & About** tab in the **Settings** of the official <Riot.im> web client.
+To use this connector you will need to have a Matrix account, and login using either your Matrix username (mxid) and password or your access token. You can find the access token in the **Advanced** section of the **Help & About** tab in the **Settings** of the official <Riot.im> web client. 
+If you use your mxid and password or if your homeserver is matrix.org, the homeserver field is optional.
 
 ## Configuration
 
 ```yaml
 connectors:
   matrix:
     # Required
-    'token': 'mytoken'  # Your matrix Access Token
-    # Alternatively, you could use your matrix username and password
     'mxid' : '@username:matrix.org'
     'password': 'mypassword'
-    # A dictionary of rooms to connect to
-    # One of these have to be named 'main'
+    # Alternatively, you could use your matrix access token
+    'token': 'mytoken'
+    # A dictionary of multiple rooms
+    # One of these should be named 'main'
     rooms:
       'main': '#matrix:matrix.org'
       'other': '#riot:matrix.org'
+    homeserver: "https://matrix.org" # Optional if you've specified mxid and password or if the homeserver is https://matrix.org
```

Comment:
We haven't actually implemented well-known support yet, so it's only optional if the server is matrix.org

Reply:
Shoud we make it optional and just explain that it defaults to matrix.org if not specified?

Reply:
tbh I would rather make it always required, but that's a bit of a breaking change. Let's document it as if it were required in the user docs when specifying a username and password, but leave the default for now. For token it should always be required.